app/assets/javascripts/search.js

```diff
@@ -0,0 +1,17 @@
+function searchIdeasListener(){
+  $("#search-box").keyup(function(){
+    var $searchInput = $('.input input').val().toLowerCase();
+    var $ideas        = $("#recent-ideas").children()
+    $ideas.hide()
+
+    var visibleIdeas = $ideas.filter(function(){
```

Comment:
So, hiding all of them and showing the ones that pass your test is a valid way of doing, but you may also want to consider looking to jQuery's `toggle` method. 

https://api.jquery.com/toggle/

Basically you could pass a boolean to `toggle()` and it if true, it will show them and if it's false, it will hide them.